pkg/gardenadm/cmd/init/init.go

```diff
@@ -143,6 +143,11 @@ func run(ctx context.Context, opts *Options) error {
 			Fn:           b.WaitUntilExtensionControllerInstallationsHealthy,
 			Dependencies: flow.NewTaskIDs(deployExtensionControllers),
 		})
+		deployNetworkPolicies = g.Add(flow.Task{
+			Name:         "Deploying network policies",
+			Fn:           b.ApplyNetworkPolicies,
+			Dependencies: flow.NewTaskIDs(deployGardenerResourceManager, deployExtensionControllers),
+		})
```

Comment:
I understand the dependencies in the way that we want to ensure that this is called after the extension namespaces are created. However, I am not sure if this is sufficient. We just wait for the deployment to finish, which essentially creates managed resources. Afterwards, `gardener-resource-manager` would need to actually apply the content.
Do you think that this is a real race condition or just nit-picking?

Second potential issue is that no other task depends on deployment of the network policies. Hence, the flow might run until after `core-dns` deployment and then abort while not yet having deployed the network policies. The next run would try to directly put `gardener-resource-manager` and extensions into the pod network, but without the network policies this will hang.
WDYT?

Reply:
> I understand the dependencies in the way that we want to ensure that this is called after the extension namespaces are created. However, I am not sure if this is sufficient. We just wait for the deployment to finish, which essentially creates managed resources. Afterwards, `gardener-resource-manager` would need to actually apply the content.
> Do you think that this is a real race condition or just nit-picking?

The namespaces for the extensions are not deployed by `gardener-resource-manager` (via `ManagedResource`) but via the `ControllerInstallation` reconciler. Hence, just depending on the deployment should be enough, no?

Reply:
Yes, that makes sense. Thanks for the clarification.